doc/neps/nep-XXXX-SIMD-optimizations.rst

```diff
@@ -0,0 +1,199 @@
+===============================================================
+NEP XXXX — Using SIMD optimization instructions for performance
+===============================================================
+
+:Author: Sayed Adel, Matti Picus, Ralf Gommers
+:Status: Draft
+:Type: Standards
+:Created: 2019-11-25
+:Resolution: none
+
+
+Abstract
+--------
+
+While compilers are getting better at using hardware-specific routines to
+optimize code, they sometimes do not produce optimal results. Also, we would
+like to be able to copy binary C-extension modules from one machine to another
+without recompiling.
+
+We have a mechanism in the ufunc machinery to choose optimal loops. This NEP
+proposes a mechanism to build on that for many more features and architectures.
+The steps would be:
+
+- Establish a baseline of CPU features for minimal support
+- Write explicit code to take advantage of well-defined, architecture-agnostic,
+  universal intrisics which capture features available across architectures.
+- Capture those universal intrisics in a set of C macros that at compile time
+  would build code paths for each feature from the baseline up to the maximum
+  set of features available on that architecture.
+- At runtime, discover which CPU features are available, and choose from among
+  the possible code paths accordingly.
+
+
+Motivation and Scope
+--------------------
+
+Traditionally NumPy has counted on the compilers to generate optimal code.
+Recently there were discussions around whether this is `good enough`_ or
```

Comment:
I don't think this was Julian's point. He's in favour of "faster", but only if it's actually faster. So a rephrase along the lines of: "Recently there were discussions about the effectiveness of adding more intrinsics (e.g., gh-xxxx for AVX optimizations for floats). There are at least two issues: benchmarking such code is hard, and build-time selection means the optimizations are mostly unused because users rely on binaries that don't contain those optimizations. This proposed design aims to solve both those issues.

Reply:
restated